File: LeetCode/01572. Matrix Diagonal Sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        output, first, last = 0, 0, len(mat[0])-1
        for index, array in enumerate(mat):
            output += array[last-index]
            if first+index != last-index:
                output += array[first+index]
        return output

File: LeetCode/01572. Matrix Diagonal Sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int output = 0;
        int first = 0;
        int last = mat[0].length-1;

        for (int index=0; index < mat.length; index++){
            output += mat[index][last-index];
            if (first+index != last-index){
                output += mat[index][first+index];
            }
        }
        return output;
    }
}
